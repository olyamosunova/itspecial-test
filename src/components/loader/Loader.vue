<template>
    <div class="loader">
        <img :src="spinner" alt="Loading..."/>
    </div>
</template>

<script lang="ts">
    import Vue from 'vue';
    import spinner from '@/assets/spinner.gif';

    export default Vue.extend({
        name: "Loader",
        data() {
            return {
                spinner
            }
        }
    })
</script>

<style scoped>
    .loader {
        flex-grow: 1;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        padding: 15px;
    }
</style>
