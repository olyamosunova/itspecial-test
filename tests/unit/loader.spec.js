import { mount, createLocalVue } from "@vue/test-utils";
import Loader from "@/components/loader/Loader";

const localVue = createLocalVue();

describe('unit tests for Loader component', () => {
    it('should match the snapshot', () => {
        const wrapper = mount(Loader, { localVue });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
