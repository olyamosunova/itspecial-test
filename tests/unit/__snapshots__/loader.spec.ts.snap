// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for Loader component should match the snapshot 1`] = `
<div class="loader d-flex justify-content-center">
  <div role="status" class="spinner-border"><span class="visually-hidden">Loading...</span></div>
</div>
`;
